File: 1859-sorting-the-sentence/1859-sorting-the-sentence.java
class Solution {
    public String sortSentence(String s) {
        String[] words = s.split(" ");
        String[] ch = new String[words.length];
        for(int i=0;i<words.length;i++){
            int index=words[i].charAt(words[i].length()-1)-'0';
            ch[index-1]=words[i].substring(0,words[i].length()-1);
        }
        return String.join(" ",ch);
    }
}

File: 1859-sorting-the-sentence/1859-sorting-the-sentence.js
/**
 * @param {string} s
 * @return {string}
 */
var sortSentence = function(s) {
  let s1=s.split(' ');
  // console.log(s1)
  let s2=[];
  for(let i=0;i<s1.length;i++){
    s2[Number(s1[i].charAt(s1[i].length-1))-1]=s1[i].replace(s1[i].charAt(s1[i].length-1),'')
  }
  // console.log(s1[0].charAt(s1[0].length-1))
  return s2.join(' ')
};